src/wireviz/wv_html.py

```diff
@@ -2,52 +2,83 @@
+        # if relative path to template was provided, check directory of YAML file first, fall back to built-in template directory
+        templatefile = smart_file_resolve(f'{metadata["template"]["name"]}.html', [Path(filename).parent, Path(__file__).parent / 'templates'])
+    else:
+        # fall back to built-in simple template if no template was provided
+        templatefile = Path(__file__).parent / 'templates/simple.html'
+
+    with open(templatefile, 'r') as file:
+        html = file.read()
+
+    # embed SVG diagram
+    with open(f'{filename}.svg') as file:
+        svgdata = file.read()
+        svgdata = re.sub(
+                  '^<[?]xml [^?>]*[?]>[^<]*<!DOCTYPE [^>]*>',
+                  '<!-- XML and DOCTYPE declarations from SVG file removed -->',
+                  svgdata, 1)
+    html = html.replace('<!-- %diagram% -->', svgdata)
+
+    # generate BOM table
+    bom = flatten2d(bom_list)
+
+    # generate BOM header (may be at the top or bottom of the table)
+    bom_header_html = '  <tr>\n'
+    for item in bom[0]:
+        th_class = f'bom_col_{item.lower()}'
+        bom_header_html = f'{bom_header_html}    <th class="{th_class}">{item}</th>\n'
+    bom_header_html = f'{bom_header_html}  </tr>\n'
+
+    # generate BOM contents
+    bom_contents = []
+    for row in bom[1:]:
+        row_html = '  <tr>\n'
+        for i, item in enumerate(row):
+            td_class = f'bom_col_{bom[0][i].lower()}'
+            row_html = f'{row_html}    <td class="{td_class}">{item}</td>\n'
+        row_html = f'{row_html}  </tr>\n'
+        bom_contents.append(row_html)
+
+    bom_html = '<table>\n' + bom_header_html + ''.join(bom_contents) + '</table>\n'
+    bom_html_reversed = '<table>\n' + ''.join(list(reversed(bom_contents))) + bom_header_html + '</table>\n'
+
+    # insert BOM table
+    html = html.replace('<!-- %bom% -->', bom_html)
+    html = html.replace('<!-- %bom_reversed% -->', bom_html_reversed)
+
+    # insert other metadata
+    if metadata:
+
+        html = html.replace('<!-- %generator% -->', f'{APP_NAME} {__version__} - {APP_URL}')
+        html = html.replace(f'"sheetsize_default"', '"{}"'.format(metadata.get('template',{}).get('sheetsize', ''))) # include quotes so no replacement happens within <style> definition
+
+        # TODO: handle multi-page documents
+        html = html.replace('<!-- %sheet_current% -->', 'Sheet<br />1')
+        html = html.replace('<!-- %sheet_total% -->', 'of 1')
+
+        # fill out other generic metadata
+        for item, value in metadata.items():
+            if isinstance(value, (str, int, float)):
+                html = html.replace(f'<!-- %{item}% -->', value)
+            elif isinstance(value, Dict):  # useful for authors, revisions
+                for index, (category, entry) in enumerate(value.items()):
+                    if isinstance(entry, Dict):
+                        html = html.replace(f'<!-- %{item}_{index+1}% -->', str(category))
```

Comment:
I am using `{index+1}` instead of `{category}` here to be able to use more generic labels in the HTML template.

This way, for example, `<!-- %revisions_1_changelog% -->` and `<!-- %authors_1_name% -->` are always replaced correctly, no matter if the user tracks revisions using numbers, letters (A, B, C, ...) or other, or if they use a different authorsship scheme instead of the `Created` and `Approved` that I have used in my examples.